tests/lib/engine.js

```diff
@@ -1,44 +1,60 @@
 var MicroEE = require('microee')
 var log = require('minilog')('test')
 
-function Socket () {
+function State () {
   this._written = []
 }
 
+var current = new State()
+
+MicroEE.mixin(State)
+
+var sockets = []
+function Socket () {
+  sockets.push(this)
+}
+
 MicroEE.mixin(Socket)
 
 Socket.prototype.sendPacket = function (nop, data) {
   var message = JSON.parse(data)
   current._written.push(message)
```

Comment:
Current provides a list of messages even through a socket reconnection. The changes in the mock engine make it possible to create more than one socket. Earlier code was pretty silly and reused the same socket again and again.